EIPS/eip-IMPERSONATECALL.md

```diff
@@ -0,0 +1,94 @@
+---
+eip: <undefined number>
+title: IMPERSONATECALL Opcode
+author: Sergio Demian Lerner (sergio.d.lerner@gmail.com)
+category: Core
+type: Standards Track
+status: Draft
+created: 2020-09-24
+---
+
+### Overview
+
+Add a new opcode, `IMPERSONATECALL` at `0xf6`, which is similar in idea to `CALL`, except that it impersonates a sender, i.e. the callee sees a sender different from the real caller. To prevent collisions with other deployed contract or externally owned accounts, the impersonated sender address is derived from the real caller address and a salt.
+
+### Specification
+
+`IMPERSONATECALL`: `0xf6`, takes 7 operands:
+
+- `gas`: the amount of gas the code may use in order to execute;
+- `to`: the destination address whose code is to be executed;
+- `in_offset`: the offset into memory of the input;
+- `in_size`: the size of the input in bytes;
+- `out_offset`: the offset into memory of the output;
+- `out_size`: the size of the scratch pad for the output.
+- `salt` is a `32` bytes value (a stack item). 
+
+#### Computation of impersonated sender
+
+The impersonated sender address is computed as `keccak256( 0xff ++ address ++ salt ++ zeros32)[12:]`.
+
+- `0xff` is a single byte, 
+- `address` is always `20` bytes, and represents the address of the real caller contract.
+- `salt` is always `32` bytes. 
+
+- The field zeros32 corresponds to 32 zero bytes.
+
+This scheme emulates `CREATE2` derivation but it cannot practically collude with the `CREATE2` address space.
+
+#### Notes
+- The opcode behaves exactly as CALL in terms of gas consumption.
+- In the called context `CALLER` returns the impersonated address.
+- If value transfer is non-zero in the call, the value is transferred from the impersonated account, and not from the real caller. This can be used to transfer ethers out of an impersonated account.
```

Comment:
```suggestion
- If value transfer is non-zero in the call, the value is transferred from the impersonated account, and not from the real caller. This can be used to transfer ether out of an impersonated account.
```